_posts/2018-11-27-pool-features.md

```diff
@@ -0,0 +1,149 @@
+- (2) If the feature is disabled it does not matter if it is supported or not. It has no effect.
+
+## `CKSUM_2K` dependant features
+
+An initial 4K of a pool is reserved for a pool header. The pool header is checksummed to control its consistency. If `CKSUM_2K` feature is enabled only first 2K of the pool header is checksummed otherwise a whole 4K. When only a first 2K of a pool header is checksummed the other 2K can be used used to store more dynamically changing data without affecting pool header consistency. Currently, the only feature which makes use of this extra space is `SHUTDOWN_STATE` feature.
+
+If the **F** feature requires to store additional information in the non-checksummed 2K of a pool header it requires to enable `CKSUM_2K` feature before enabling the **F** feature. On the other hand, before `CKSUM_2K` feature is disabled it is required to disable the **F** feature if the **F** feature requires to store additional information in the non-checksummed 2K of a pool header.
+
+# Feature toggling and querying
+
+Toggling and querying features for already existing and **NOT** opened pools can be performed using the `pmempool` tool:
+
+```bash
+$ pmempool feature --enable SHUTDOWN_STATE /path/to/poolset.file # to enable
+
+$ pmempool feature --disable SHUTDOWN_STATE /path/to/poolset.file # to disable
+
+$ pmempool feature --query SHUTDOWN_STATE /path/to/poolset.file # to query
+0 # query result: 0 == disabled, 1 == enabled
+```
+
+or using *libpmempool* API calls:
+
+```c
+const char *path = "/path/to/poolset.file";
+enum pmempool_feature f = SHUTDOWN_STATE;
+unsigned flags = 0; /* only currently supported value */
+int result = 0;
+
+/* to enable */
+result = pmempool_feature_enable(path, f, flags);
+if (result < 0) {
+	fprintf(stderr, "pmempool_feature_enable failed with errno: %d", errno);
+}
+
+/* to disable */	
+result = pmempool_feature_disable(path, f, flags);
+if (result < 0) {
+	fprintf(stderr, "pmempool_feature_disable failed with errno: %d", errno);
+}
+
+/* to query */
+result = pmempool_feature_query(path, f, flags);
+if (result < 0) {
+	fprintf(stderr, "pmempool_feature_query failed with errno: %d", errno);
+} else {
+	printf("pmempool_feature_query results is %s", result == 0 ? "disabled" : "enabled");
+}
+```
+
+**Disclaimers**:
+
+- toggling features is not supported for poolsets with remote replicas
+- `SINGLEHDR` can only be queried because it has to be explicitly written in a [poolset file][poolset.5]
+
+## Set `SHUTDOWN_STATE` initial value
+
+Every feature has its initial value either enabled or disabled. If it is required to change the feature value after a pool is created it can be done as it is described in [Feature toggling and querying](#feature-toggling-and-querying). But sometimes it is useful to change an initial feature value to have enabled or disabled feature from the get-go.
```

Comment:
Some features might be disabled by default. To enable them at creation time, use the appropriate query from the [CTL namespace]().